[OCPCLOUD-921] add alert for machine with long deletion phase

elmiko opened the pull request:
This change adds the alert for a machine with phase "Deleting" for
longer than 180 minutes, at "warning" severity. Also updates the alerts
documentation with new information.

The MachineWithNoRunningPhase alert has also been modified to exclude
machines with the "Deleting" phase to ensure that we do not double fire
an alert for this condition.

ref: https://issues.redhat.com/browse/OCPCLOUD-921

JoelSpeed reviewed (commented):
I think this looks fine, would be good to get an SRE opinion on it before we merge though

RiRa12621 commented:
@JoelSpeed LGTM from our end.
[x] Documentation exists
[x] Root cause based alert does not exceed warning level

RiRa12621 commented:
/assign @RiRa12621 
assign to myself so it's a bit easier for me find this later

JoelSpeed commented:
/approve
/retest

Thanks for the review @RiRa12621

elmiko commented:
just realized i set this for 3hours, but we specified 6hours in the enhancement. i'm going to bump it up.

michaelgugino commented on docs/user/Alerts.md:
````diff
@@ -33,6 +33,25 @@ Machine did not reach the “Running” Phase.  Running phase is when the machin
+(mapi_machine_created_timestamp_seconds{phase=="Deleting"}) > 0
+```
+
+### Possible Causes
````
Add: A pod with a very long GracePeriodTimeout has not stopped yet

michaelgugino commented on docs/user/Alerts.md:
```diff
@@ -33,6 +33,25 @@ Machine did not reach the “Running” Phase.  Running phase is when the machin
+  preventing Node removal.
+
+### Resolution
+Consult the `machine-controller`'s logs for root causes (see the [Troubleshooting Guide](TroubleShooting.md). In some
```
We should work to ensure in the future that looking at the logs is not a necessity.  We should prioritize putting the information on the machine's status.

michaelgugino reviewed (commented):
A couple small things, otherwise looks good to me.

elmiko commented:
/hold
i need to fix up a few thing

2uasimojo commented:
👀

elmiko commented:
updated the pr:
* changed error name to MachineNotYetDeleted
* include a mention of graceful termination period in possible causes
* update alert timing to 360 minutes

elmiko commented:
/hold cancel

dofinn commented:
The alert itself LGTM
@RiRa12621 @jeremyeder is this a candidate for https://github.com/openshift/runbooks ?

RiRa12621 commented:
Didn't see you ping @dofinn 
yes it is, but we're not super clear yet on the structure. I'll port this one over once we're ready with the runbooks repo

JoelSpeed commented:
/retest

@dofinn Are you happy for us to merge this into 4.8 or will that cause issues for you this late?

RiRa12621 commented:
lgtm from my end @JoelSpeed 
as for the runbooks dom mentioned, we're still bootstrapping that bit, so don't worry about it for now

JoelSpeed commented:
/lgtm

JoelSpeed commented:
/override ci/prow/e2e-aws-workers-rhel7

This job no longer exists and so will never be able to pass

michaelgugino reviewed (approved):
/approve

JoelSpeed commented:
This is a feature, so we need to wait until master opens before we can merge it. Discussed already with @elmiko, if we leave this without hold, it should merge as soon as master reopens

RiRa12621 commented:
Are we still confident this will happen automatically?

elmiko commented:
> Are we still confident this will happen automatically?

i want to believe that, but if not we will kick the CI machine to merge it ;)

RiRa12621 commented:
Sounds perfect. Was just wondering since it's been a month an I wasn't sure if master had opened meanwhile.

elmiko commented:
it looks like 4.9 development will open on June 11.

stbenjam commented:
/retitle Bug 1974758: [OCPCLOUD-921] add alert for machine with long deletion phase
